src/pages/iou/request/step/IOURequestStepRate.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import SelectionList from '@components/SelectionList';
+import RadioListItem from '@components/SelectionList/RadioListItem';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as IOU from '@libs/actions/IOU';
+import compose from '@libs/compose';
+import DistanceRequestUtils from '@libs/DistanceRequestUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Route} from '@src/ROUTES';
+import type {Policy} from '@src/types/onyx';
+import StepScreenWrapper from './StepScreenWrapper';
+import withWritableReportOrNotFound from './withWritableReportOrNotFound';
+
+type Props = {
+    /** Object of last selected rates for the policies */
+    lastSelectedDistanceRates: Record<string, string>;
+
+    /** Policy details */
+    policy: OnyxEntry<Policy>;
+
+    /** The route object passed to this screen */
+    route: {
+        /** The params passed to this screen */
+        params: {
+            /** The route to go back to */
+            backTo: Route;
+        };
+    };
+};
+
+function IOURequestStepRate({
+    policy,
+    route: {
+        params: {backTo},
+    },
+    lastSelectedDistanceRates = {},
+}: Props) {
+    const styles = useThemeStyles();
+    const {translate, toLocaleDigit} = useLocalize();
+    const rates = DistanceRequestUtils.getMileageRates(policy?.id);
+
+    const lastSelectedRate = lastSelectedDistanceRates[policy?.id ?? '0'] ?? rates[0]?.customUnitRateID;
+
+    const data = Object.values(rates).map((rate) => ({
+        text: rate.name ?? '',
+        alternateText: DistanceRequestUtils.getRateForDisplay(true, rate.unit, rate.rate, rate.currency, translate, toLocaleDigit),
+        keyForList: rate.name ?? '',
+        value: rate.customUnitRateID,
+        isSelected: lastSelectedRate ? lastSelectedRate === rate.customUnitRateID : Boolean(rate.name === CONST.CUSTOM_UNITS.DEFAULT_RATE),
+    }));
+
+    const unit = Object.values(rates)[0]?.unit === CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES ? translate('common.mile') : translate('common.kilometer');
+
+    const initiallyFocusedOption = rates[lastSelectedRate]?.name ?? CONST.CUSTOM_UNITS.DEFAULT_RATE;
+
+    function selectDistanceRate(customUnitRateID: string) {
+        IOU.setLastSelectedDistanceRates(policy?.id ?? '', customUnitRateID);
+        IOU.updateDistanceRequestRate('1', customUnitRateID);
```

Comment:
Let's move setLastSelectedDistanceRates into updateDistanceRequestRate. Conceptually I feel like they're one action.

Reply:
as long as we need this logic https://github.com/Expensify/App/pull/38543/files#r1553730943, they should be separate

Reply:
Sorry, that link isn't working for me. This is the only use of setLastSelectedDistanceRates I see in the code, so can we combine it?